ui/components/app/wallet-overview/coin-overview.tsx

```diff
@@ -91,19 +114,55 @@ export const CoinOverview = ({
   const metaMetricsId = useSelector(getMetaMetricsId);
   const isMetaMetricsEnabled = useSelector(getParticipateInMetaMetrics);
   const isMarketingEnabled = useSelector(getDataCollectionForMarketing);
+
   ///: END:ONLY_INCLUDE_IF
 
-  const isEvm = useSelector(getMultichainIsEvm);
-  const showFiat = useSelector(getMultichainShouldShowFiat);
-  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
-  const { ticker, type, rpcUrl } = useSelector(getMultichainProviderConfig);
-  const isOriginalNativeSymbol = useIsOriginalNativeTokenSymbol(
-    chainId,
-    ticker,
-    type,
-    rpcUrl,
+  const showNativeTokenAsMainBalanceRoute = getSpecificSettingsRoute(
+    t,
+    t('general'),
+    t('showNativeTokenAsMainBalance'),
   );
+  const theme = useTheme();
+
+  const shouldShowPopover = useSelector(getShouldShowAggregatedBalancePopover);
+  const isTestnet = useSelector(getIsTestnet);
+  const { showFiatInTestnets } = useSelector(getPreferences);
+
+  const selectedAccount = useSelector(getSelectedAccount);
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+  const { totalFiatBalance } = useAccountTotalFiatBalance(
+    selectedAccount,
+    shouldHideZeroBalanceTokens,
+  );
+
+  const { showNativeTokenAsMainBalance } = useSelector(getPreferences);
+
+  const isEvm = useSelector(getMultichainIsEvm);
+  const balanceToDisplay =
+    showNativeTokenAsMainBalance || isTestnet || !isEvm
+      ? balance
+      : totalFiatBalance;
+
   const tokensMarketData = useSelector(getTokensMarketData);
+  const [isOpen, setIsOpen] = useState(true);
+
+  const handleMouseEnter = () => {
+    setIsOpen(true);
+  };
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+    setAggregatedBalancePopover();
+  };
+
+  const [referenceElement, setReferenceElement] = useState();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const setBoxRef = (ref: any) => {
+    setReferenceElement(ref);
+  };
```

Comment:
Will this fix the `no-explicit-any` thing and still retain functionality?

```suggestion
  const [referenceElement, setReferenceElement] =
    useState<HTMLSpanElement | null>(null);

  const setBoxRef = (ref: HTMLSpanElement | null) => {
    if (ref) {
      setReferenceElement(ref);
    }
  };
```

Reply:
Thnx for pointing that out 🙏